docs/tutorials/cooperative_groups_tutorial.rst

```diff
@@ -0,0 +1,312 @@
+      }
+
+      // Only the first thread returns a valid value
+      if(g.thread_rank() == 0)
+          return val;
+      else
+          return 0;
+  }
+
+The ``reduce_sum`` device function reused to calculate the block and custom
+partition sum of the input numbers. The kernel has three different sections:
+1. The reduction function variables initialization.
+2. The reduction of thread block and store the results in global memory.
+3. The reduction of custom partition and store the results in global memory.
+
+
+1. The reduction function variables initialization
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In this code section the shared memory is declared, the thread_block_group and
+custom_partition are defined and the input variable are loaded from global
+memory.
+
+.. code-block:: cpp
+
+  // threadBlockGroup consists of all threads in the block
+  thread_block thread_block_group = this_thread_block();
+
+  // Workspace array in shared memory required for reduction
+  __shared__ unsigned int workspace[2048];
+
+  unsigned int output;
+
+  // Input to reduce
+  const unsigned int input = d_vector[thread_block_group.thread_rank()];
+
+  // ...
+
+  // Every custom_partition group consists of 16 threads
+  thread_block_tile<PartitionSize> custom_partition
+          = tiled_partition<PartitionSize>(thread_block_group);
+
+
+2. The reduction of thread block
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In this code section the sum is calculate on thread_block_group level, then the
+results are stored in global memory.
+
+.. code-block:: cpp
+
+  // Perform reduction
+  output = reduce_sum(thread_block_group, workspace, input);
+
+  // Only the first thread returns a valid value
+  if(thread_block_group.thread_rank() == 0)
+  {
+    d_block_reduced_vector[0] = output;
+  }
+
+3. The reduction of custom partition
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In this code section the sum is calculate on thread_block_group level, then the
+results are stored in global memory.
+
+.. code-block:: cpp
+
+  // Perform reduction
+  output = reduce_sum(custom_partition, &workspace[group_offset], input);
+
+  // Only the first thread in each partition returns a valid value
+  if(custom_partition.thread_rank() == 0)
+      {
+          const unsigned int partition_id          = thread_block_group.thread_rank() / PartitionSize;
+          d_partition_reduced_vector[partition_id] = output;
+      }
+
+Host side code
+--------------
+
+At the host side, you have to do the following steps:
+1. Check the cooperative group support on AMD GPUs.
+2. Initialize the cooperative group configuration.
+3. Allocate and copy input to global memory.
+4. Launch the cooperative kernel.
+5. Save the results from global memory.
+6. Free the global memory.
+
+1. Check the cooperative group support on AMD GPUs
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Not all the GPU supports cooperative groups, to make sure you should check with
+the following code:
```

Comment:
Not all AMD GPUs support cooperative groups. You can confirm support with the following code:

Reply:
Fixed.